### PR TITLE
[ENH] add a difference transformer to series transformations

### LINKED ISSUE #1553: [ENH] `Differencer` series transformer
### Describe the feature or idea you want to propose

The previous transformer for finding series difference (i.e. first-order differences, second-order differences etc.) is deprecated as it was overcomplicated and used the old interface.

It would be good to have a simpler version in the new interface.

### Describe your proposed solution

A differencer extending `BaseSeriesTransformer`.

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by TinaJin0228**
#### Reference Issues/PRs
fixes #1553 

#### What does this implement/fix? Explain your changes.
This implements a transformer that computes the n-order differences of a time series. 

The input time series is expected to have a NumPy inner-type, and the transformation is performed using numpy.diff(). 

To preserve the original shape of the series along the time axis, the first order element(s) of the output are filled with NaN.

#### Does your contribution introduce a new dependency? If yes, which one?
No. It only depends on NumPy.

#### Any other comments?
No.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you **after** the PR has been merged.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [ ] I've added the estimator/function to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).
- [ ] (OPTIONAL) I've added myself as a `__maintainer__` at the top of relevant files and want to be contacted regarding its maintenance. Unmaintained files may be removed. This is for the full file, and you should not add yourself if you are just making minor changes or do not want to help maintain its contents.

##### For developers with write access
- [ ] (OPTIONAL) I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

**Comment by TinaJin0228** [comment]
It seems that one of the automated checks failed with errors indicating it couldn't find the merge base or my commit SHA: 
```
Warning: Unable to find merge base between 45d0390d1a441264943f8963547409bb36935d2d and 8f78ec98e57b49640bd59f5570f35efb9fea7e67
Error: Unable to locate the commit sha: 45d0390d1a441264943f8963547409bb36935d2d
Error: Please verify that the commit sha is correct, and increase the 'fetch_depth' input if needed
Warning: fatal: bad object 45d0390d1a441264943f8963547409bb36935d2d
Warning: If this pull request is from a forked repository, please set the checkout action `repository` input to the same repository as the pull request.
Warning: This can be done by setting actions/checkout `repository` to ${{ github.event.pull_request.head.repo.full_name }}
```

File: aeon/transformations/series/_diff.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        "capability:multivariate": True,
+        "X_inner_type": "np.ndarray",
+        "fit_is_empty": True,
+    }
+
+    def __init__(self, order=1, axis=1):
+        if not isinstance(order, int) or order < 1:
+            raise ValueError(f"`order` must be a positive integer, but got {order}")
+        self.order = order
+        super().__init__(axis=axis)
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/series/_diff.py:82]: Remove the `axis` parameter. This is used to determine the shape of the time series internally. We only want to apply this to series, not between channels.

File: aeon/transformations/series/_diff.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    _tags = {
+        "capability:multivariate": True,
+        "X_inner_type": "np.ndarray",
+        "fit_is_empty": True,
+    }
+
+    def __init__(self, order=1, axis=1):
+        if not isinstance(order, int) or order < 1:
+            raise ValueError(f"`order` must be a positive integer, but got {order}")
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/series/_diff.py:80]: I would do this in `fit`

File: aeon/transformations/series/tests/test_diff.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+from aeon.transformations.series._diff import DifferenceTransformer
+
+
+def test_diff():
+    """Tests basic first and second order differencing."""
+    X = np.array([[1.0, 4.0, 9.0, 16.0, 25.0, 36.0]])
+
+    dt1 = DifferenceTransformer(order=1)
+    Xt1 = dt1.fit_transform(X)
+    expected1 = np.array([[np.nan, 3.0, 5.0, 7.0, 9.0, 11.0]])
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/series/tests/test_diff.py:14]: IMO better to return a smaller series than include NaNs. Possibly a parameter if you think it is worth it but by default change the shape.

File: aeon/transformations/series/tests/test_diff.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+"""Tests for Difference transformation."""
+
+import numpy as np
+
+from aeon.transformations.series._diff import DifferenceTransformer
+
+
+def test_diff():
+    """Tests basic first and second order differencing."""
+    X = np.array([[1.0, 4.0, 9.0, 16.0, 25.0, 36.0]])
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/series/tests/test_diff.py:10]: Can you also test multivariate series in another test perhaps

**Comment by TinaJin0228** [comment]
I have made som modifications according to your comments. @MatthewMiddlehurst

File: aeon/transformations/series/_diff.py (hunk 1 above)
Comment (TinaJin0228) [reply aeon/transformations/series/_diff.py]: got it

**Comment by TinaJin0228** [comment]
> I have made som modifications according to your comments. @MatthewMiddlehurst

@MatthewMiddlehurst I am wondering if you could give it a quick look when you get a chance? Thanks!

**Review by TonyBagnall — APPROVED** [review]
lgtm, thanks

File: aeon/transformations/series/_diff.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
+import numpy as np
+
+from aeon.transformations.series.base import BaseSeriesTransformer
+
+__maintainer__ = ["Tina Jin"]
```
Comment (TonyBagnall) [inline aeon/transformations/series/_diff.py:5]: could you put your github handle here?

Comment (TinaJin0228) [reply aeon/transformations/series/_diff.py:5]: fine

**Review by MatthewMiddlehurst — APPROVED** [review]
